### PR TITLE
Examples/angular router

### LINKED ISSUE #42: ensure compatibility of `urlSync: true` with angular router

### PR DESCRIPTION
**Pull request by iam4x**
fixes #42

But we need https://github.com/algolia/instantsearch.js/pull/2604 and new InstantSearch.js beta first.

### CODE REVIEW

**Comment by iam4x** [comment]
Deploy preview failed.

Built with commit 1348e80e41fc5d7a93eebadfde5662bccb0dedef

https://app.netlify.com/sites/angular-instantsearch/deploys/5a1c3190a114770c50e5b6df

File: package.json [hunk 1]
```diff
@@ -14,8 +14,10 @@
     "test": "jest",
     "test:ci": "yarn lint && jest --runInBand && yarn build",
     "test:watch": "jest --watch",
-    "examples:ecommerce": "./scripts/run-ecommerce.sh",
-    "examples:ecommerce:build": "./scripts/build-ecommerce.sh",
+    "examples:ecommerce": "./scripts/run-example.sh e-commerce",
+    "examples:ecommerce:build": "./scripts/build-example.sh e-commerce",
```
Comment (Haroenv) [inline package.json]: The build script wasn't renamed to `build-example` @iam4x

Comment (iam4x) [reply package.json]: Yes I did not updated it yet :)

**Comment by vvo** [comment]
So how does urlSync and angular router are working together? Do you have an API much like RIS where you can plug-in any routing mechanism and inject state?

**Comment by iam4x** [comment]
* They are working well together, nothing to change.
* What are the benefits to plug it with the Angular router mechanism itself? Right now it updates on its own the URL query to keep the state synced, like IS.js.

**Comment by iam4x** [comment]
Demo is here: https://5a1d3497a6188f471d8c67c7--angular-instantsearch.netlify.com/angular-router

It's currently broken, it will be fixed with the new InsantSearch.js 2.3 beta version 👍

**Comment by Haroenv** [comment]
Is it possible that someone adds query strings with angular router and it conflicts with this option, is what @vvo meant I think

**Comment by iam4x** [comment]
If he uses an already used query param from the urlSync mechanism, yes it will conflict and experience weird behaviours.

In any other cases, should be fine, the urlSync mechanism will just merge every query params and do not replace the ones added by the user 👍

**Comment by iam4x** [comment]
@vvo Yes exactly, in fact the angular router can access the query params but it does not have methods to set query string. Everything is declarative and uses `<a>` links to update the query.

---

Ran into a new issue, PR submitted on is.js: https://github.com/algolia/instantsearch.js/pull/2610

**Comment by vvo** [comment]
🍾

**Pull request merged**